src/kafka/base.rs

```diff
@@ -163,8 +163,34 @@ pub trait AlertProducer {
             )?;
             debug!(?total_messages);
 
-            // TODO: need to get the expected count
-            // TODO: finish the produce-or-not logic
+            // Count the number of Avro files in the data directory
+            let avro_count = match count_files_in_dir(&self.data_directory(), Some(&["avro"])) {
+                Ok(count) => count,
+                Err(_) => 0, // If we can't count files, the directory must not exist
+            };
+
+            // If they don't match, delete all messages from the topic and
+            // start fresh
+            if total_messages > 0 && total_messages != avro_count as u32 {
+                warn!(
+                    "Topic {} already exists with {} messages, but {} Avro files found in data directory",
+                    self.topic_name(),
+                    total_messages,
+                    avro_count
+                );
+                // TODO: Delete all messages from Kafka topic so we start fresh
+                // For now, just return an error
+                return Err(
```

Comment:
Oh I didn't realize that's the bahavior you guys had in mind? I thought we wanted to skip pushing alerts when the count matches and otherwise wipe the topic and push again, but here it seems like you want to alert the user and have them clear the topic manually?

If we don't want to automate the topic deletion that's fine, but I think that providing the users with a `--clear` flag to let them delete the topic and push again using boom rather than dive into a kafka container or similar to run these commands themselves would be a LOT nicer.

Reply:
I was trying to write the code to clear the topic but I couldn't figure it out after trying for a bit (async return types this that and the other), so I landed on just raising an error message.